subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -51,62 +43,133 @@ enum {
 	ADV_FLAGS_NUM
 };
 
-static struct {
+typedef struct net_buf* (*adv_buf_handler_t)(k_timeout_t timeout);
+
+struct ext_adv {
 	ATOMIC_DEFINE(flags, ADV_FLAGS_NUM);
 	struct bt_le_ext_adv *instance;
 	struct net_buf *buf;
 	uint64_t timestamp;
 	struct k_work_delayable work;
-} adv;
+	adv_buf_handler_t handler;
+	struct bt_le_adv_param adv_param;
+};
 
-static int adv_start(const struct bt_le_adv_param *param,
+static void send_pending_adv(struct k_work *work);
+static struct ext_adv adv_main = {
+	.work = Z_WORK_DELAYABLE_INITIALIZER(send_pending_adv),
+	.handler = bt_mesh_adv_buf_get,
```

Comment:
Here, there is a change. The first advertising set (call `adv_main`) has the ability to send relay messages. I think this is also reasonable. When there is no local message to send, it should not choose to ignore it.